api/client/webclient/webclient.go

```diff
@@ -219,13 +220,24 @@ func Ping(cfg *Config) (*PingResponse, error) {
 		return nil, trace.Wrap(err)
 	}
 	defer resp.Body.Close()
-	if resp.StatusCode == http.StatusBadRequest {
-		per := &PingErrorResponse{}
-		if err := json.NewDecoder(resp.Body).Decode(per); err != nil {
-			return nil, trace.Wrap(err)
+
+	if resp.StatusCode != http.StatusOK {
+		slog.DebugContext(req.Context(), "Received unsuccessful ping response", "code", resp.StatusCode)
+
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, trace.Wrap(err, "could not read ping response body")
 		}
-		return nil, errors.New(per.Error.Message)
+
+		errResp := &PingErrorResponse{}
+		if err := json.Unmarshal(bodyBytes, errResp); err != nil {
+			slog.DebugContext(req.Context(), "Could not parse ping response body", "body", string(bodyBytes))
+			return nil, trace.Wrap(err, "cannot parse unsuccessful ping response")
```

Comment:
I think this may happen with load balancers etc. in the mix. Perhaps we can suggest that? For example:

```suggestion
			return nil, trace.Wrap(err, "cannot parse ping response; is proxy reachable?")
```